### PR TITLE
BFB-126: DummyTarget Behaviour

### PR DESCRIPTION
**Pull request by unoctium1**
**JIRA Ticket**
https://perigongames.atlassian.net/browse/BFB-126

**Description**
Create dummy target behavior, integrate bullets with LifeCycleBehaviour, and placeholder animations

**Screenshots / GIFs**
https://youtu.be/4hRgjpTM4PI

### CODE REVIEW

File: Assets/Plugins/RootMotion/FinalIK.meta [hunk 1]
```diff
@@ -0,0 +1,8 @@
+fileFormatVersion: 2
```
Comment (m4rtint) [inline Assets/Plugins/RootMotion/FinalIK.meta:1]: Is this for something else or...

Comment (unoctium1) [reply Assets/Plugins/RootMotion/FinalIK.meta]: Sorry about that, didn't mean to commit that!

File: Assets/_BForBoss/_AI/Scripts/DummyTargetBehaviour.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        
+        protected override void Awake()
+        {
+            base.Awake();
+            _animator = GetComponentInChildren<Animator>();
+        }
+
+        protected override void LifeCycleFinished()
+        {
+            _animator.SetBool(DEATH_ID, true);
```
Comment (m4rtint) [inline Assets/_BForBoss/_AI/Scripts/DummyTargetBehaviour.cs:26]: Do you know why a shotgun shot would call this twice? And is it a huge deal or is it just placeholder.

Comment (m4rtint) [reply Assets/_BForBoss/_AI/Scripts/DummyTargetBehaviour.cs]: to be more specific, the death animation is shown twice.

Comment (unoctium1) [reply Assets/_BForBoss/_AI/Scripts/DummyTargetBehaviour.cs]: Hmm that could be an issue later on, I'll work on a fix for that

File: Assets/_BForBoss/_Entities/Animation/Fighting Idle.fbx.meta [hunk 3]
```diff
@@ -0,0 +1,768 @@
+fileFormatVersion: 2
```
Comment (m4rtint) [inline Assets/_BForBoss/_Entities/Animation/Fighting Idle.fbx.meta:1]: Do we actually need the whole fbx? or just the animation inside?

Comment (unoctium1) [reply Assets/_BForBoss/_Entities/Animation/Fighting Idle.fbx.meta:1]: They're skinless fbx's so I don't know that it matters? I can try to extract the anim file but I'm a little scared that will break things

File: Assets/_BForBoss/_Weapons/Scripts/BulletBehaviour.ApplyDamage.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
+using Perigon.Entities;
+using UnityEngine;
+
+namespace Perigon.Weapons
+{
+    public interface IApplyDamage
+    {
+        public void HitObject(Collider col);
```
Comment (m4rtint) [inline Assets/_BForBoss/_Weapons/Scripts/BulletBehaviour.ApplyDamage.cs:8]: NIT, you don't need the `public`

File: Assets/_BForBoss/_Entities/Animation/Fighting Idle.fbx.meta (hunk 3 above)
Comment (unoctium1) [reply Assets/_BForBoss/_Entities/Animation/Fighting Idle.fbx.meta:1]: Nvm, it doesn't break things

File: Assets/_BForBoss/_Weapons/Scripts/BulletBehaviour.ApplyDamage.cs (hunk 4 above)
Comment (unoctium1) [reply Assets/_BForBoss/_Weapons/Scripts/BulletBehaviour.ApplyDamage.cs:8]: Should I just get rid of the interface all together? Wasn't sure if that was a best practice for partial classes

Comment (m4rtint) [reply Assets/_BForBoss/_Weapons/Scripts/BulletBehaviour.ApplyDamage.cs:8]: So I personally can't currently think of any place that'll need to inject the bullets into. 
So I wouldn't create an interface for this.

Comment (unoctium1) [reply Assets/_BForBoss/_Weapons/Scripts/BulletBehaviour.ApplyDamage.cs:8]: Sure I'll get rid of it then

**Review by m4rtint — APPROVED** [review]
lgtm

**Pull request merged**